### PR TITLE
Adding a DatabaseQuery action that can run after a pipeline run.

### PR DESCRIPTION
**Pull request by albertshau**
This allows people to do things like run a cleanup query at the
end of a pipeline run.

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java [hunk 1]
```diff
@@ -99,12 +99,7 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
   @Override
   public final void prepareRun(BatchSourceContext context) {
-    Map<String, String> runtimeArgs = context.getRuntimeArguments();
     long runtime = context.getLogicalStartTime();
-    if (runtimeArgs.containsKey("runtime")) {
-      runtime = Long.parseLong(runtimeArgs.get("runtime"));
```
Comment (gsps1) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java:105]: why this change ?

**Comment by gsps1** [comment]
lgtm, just one question

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java (hunk 1 above)
Comment (gsps1) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java:105]: was this a hack before ? but seems we we used this as a way to process for a particular time, but now that has to be passed by reconfiguring the pipeline?

Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java:105]: yeah this was a workaround before, but now CDAP has the ability to set 'logical.start.time' as a runtime argument to override the normal logical start time.

Comment (gsps1) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/TimePartitionedFileSetSource.java:105]: ok, got it.
